homework_multithreading_2/src/main/java/org/application/Bartender.java

```diff
@@ -0,0 +1,56 @@
+package org.application;
+
+
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.TimeUnit;
+
+public class Bartender implements Runnable {
+    private final Bar bar;
+    private volatile String drink;
+    private final int bartender;
+    private final int clients;
+
+    public Bartender(Bar bar, int bartender, int clients) {
+        this.bar = bar;
+        this.bartender = bartender;
+        this.clients = clients;
+    }
+
+    @Override
+    public void run() {
+        BlockingQueue<String> orders = bar.getOrders();
+        getOrder(orders);
+    }
+
+    private void getOrder(BlockingQueue<String> orders) {
+        for (int i = 0; i < getCountOfOrders(); i++) {
+            try {
+                String order = orders.poll(1, TimeUnit.SECONDS);
+                if (order == null) {
+                    break;
+                }
+                getString(order);
+                bar.getReadyOrders().add(prepareDrink());
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    private void getString(String order) {
+        String[] strings = order.split(" ");
+        drink = strings[strings.length - 1];
+        System.out.printf(Thread.currentThread().getName() + " received an order for brewing %s%n", drink);
+    }
+
+    private int getCountOfOrders() {
+        if (clients == bartender) return 1;
+        if (clients % bartender == 0) return clients / bartender;
+        return (clients / bartender) + 1;
```

Comment:
What if clients = 3 and bartender = 2 ?
Or clients = 1 and bartender = 5

Reply:
then everything works correctly, because the poll method is used, which waits 1 second and if the queue is empty then it finishes working